R/utils.R

```diff
@@ -1862,3 +1862,147 @@ find_rank_bias <- function(mat,
         return(rdiff)
     }
 }
+
+#' Given a reference matrix and a list of genes, take the union of 
+#' all genes in vector and genes in reference matrix 
+#' and insert zero counts for all remaining genes. 
+#' @param geneVector char vector with gene names
+#' @param ref_matrix Reference matrix containing cell types vs. 
+#' gene expression values
+#' @return Reference matrix with union of all genes
+append_genes <- function(gene_vector, ref_matrix)
+{
+    rownamesGSEMatrix <- rownames(ref_matrix) 
+    #Get rownames from GSEMatrix (new GSE file)
+    
+    rowCountHumanGenes <- nrow(gene_vector) 
+    #Calculate number of rows from list of full human genes
+    rowCountNewGSEFile <- nrow(ref_matrix) 
+    #Calculate number of rows of GSE matrix
+    
+    missing_rows <- setdiff(gene_vector, rownamesGSEMatrix) 
+    #Use setdiff function to figure out rows which are different/missing 
+    #from GSE matrix
+    
+    zeroExpressionMatrix <- matrix(
+        0, 
+        nrow = length(missing_rows), 
+        ncol = ncol(ref_matrix)) 
+    #Create a placeholder matrix with zeroes and missing_rows length
+    
+    rownames(zeroExpressionMatrix) <- missing_rows 
+    #Assign row names
+    colnames(zeroExpressionMatrix) <- colnames(ref_matrix) 
+    #Assign column names
+    
+    full_matrix <- rbind(ref_matrix, zeroExpressionMatrix) 
+    #Bind GSEMatrix and zeroExpressionMatrix together
+    
+    #Reorder matrix
+    full_matrix <- full_matrix[gene_vector, ] 
+    #Reorder fullMatrix to preserve gene order
+    return(full_matrix) 
+    #Return fullMatrix
+}
+
+#' Given a count matrix, determine if the matrix has been either 
+#' log-normalized, normalized, or contains raw counts
+#' @param counts_matrix Count matrix containing scRNA-seq read data
+#' @param max_log_value Static value to determine if a matrix is normalized
+#' @return String either raw counts, log-normalized or normalized
+#' @example
+#' mat <- append_genes(
+#' gene_vector = human_genes_10x,
+#' ref_matrix = cbmc_ref 
+#' ) 
+#' @export
+check_raw_counts <- function(counts_matrix, max_log_value = 50)
```

Comment:
apparently this does not recognize sparse matrix...
compare
```
check_raw_counts(counts_matrix = as.matrix(pbmc_matrix_small), max_log_value = 50)
check_raw_counts(counts_matrix = pbmc_matrix_small, max_log_value = 50)
```

test with something like this: 
```
if (is(pbmc_matrix_small, 'sparseMatrix')) {
  pbmc_matrix_small <- as.matrix(pbmc_matrix_small)
}
```

Reply:
See bd9647e

Reply:
Also see 4992003